src/samplers/sampler.jl

```diff
@@ -32,7 +33,7 @@ assume(spl::Void, dist::Distribution, vn::VarName, vi::VarInfo) = begin
   if haskey(vi, vn)
     r = vi[vn]
   else
-    r = rand(dist)
+    r = init(dist)
```

Comment:
But IS do have its own definition of `assume()` to generate samples (below). It always calls `model` with `sampler::Sampler{IS}`.
```julia
assume(spl::Sampler{IS}, dist::Distribution, vn::VarName, vi::VarInfo) = begin
  r = rand(dist)
  push!(vi, vn, r, dist, 0)
  r
end
```